content/200-orm/100-prisma-schema/10-overview/03-generators.mdx

```diff
@@ -55,6 +59,125 @@ If you use macOS ARM64 (`darwin-arm64`), then the binary file that was compiled
 
 > **Note**: The `native` binary target is the default. You can set it explicitly if you wish to include additional [binary targets](/orm/reference/prisma-schema-reference#binarytargets-options) for deployment to different environments.
 
+
+## New `prisma-client` generator (Preview)
+
+The new `prisma-client` generator offers greater control and flexibility when using Prisma ORM across different Node.js environments (including ECMAScript modules). 
+
+The  `prisma-client` generator generates the client **outside** of `node_modules` and inside your application code, allowing you to maintain full visibility over the generated code, including the query engine.
+
+Currently in [Preview](/orm/more/releases#preview), this generator ensures you can bundle your application code exactly the way you want, without relying on hidden or automatic behaviors.
+
+### Getting started
+
+Follow these steps to use the new `prisma-client` generator in your project.
+
+#### 1. Configure the `prisma-client` generator in `schema.prisma`
+
+Update your [`generator`](/orm/prisma-schema/overview/generators) block:
+
+```prisma file=prisma/schema.prisma
+generator client {
+  //add-start
+  provider = "prisma-client"            // Required
+  output   = "../src/generated/prisma/client"  // Required path
+  //add-end
+}
+```
+
+The **`output` option is required** and tells Prisma where to generate the client. You can choose any location suitable for your project structure. For instance, if you have the following layout:
+
+```txt
+.
+├── package.json
+├── prisma
+│   └── schema.prisma
+├── src
+│   └── index.ts
+└── tsconfig.json
+```
+
+Then `../src/generated/prisma/client` places the generated code in `src/generated/prisma/client` relative to `schema.prisma`.
+
+#### 2. Generate the Prisma Client
+
+Generate the `PrismaClient` by running:
+
+```bash
+npx prisma generate
+```
+
+This compiles the necessary TypeScript definitions and includes the query engine in the configured `output` folder.
+
+#### 3. Exclude the generated directory from version control
+
+The new generator includes both the TypeScript client code **and** the [query engine](/orm/more/under-the-hood/engines#the-query-engine-file). Including the query engine in version control can cause compatibility issues on different machines. To avoid this, add the generated directory to `.gitignore`:
```

Comment:
```suggestion
The new generator includes both the TypeScript client code _and_ the [query engine](/orm/more/under-the-hood/engines#the-query-engine-file). Including the query engine in version control can cause compatibility issues on different machines. To avoid this, add the generated directory to `.gitignore`:
```